src/blockperf/config.py

```diff
@@ -73,50 +77,142 @@ def check_blockperf_config(self):
-            "NormalVerbosity",
-            "MaximalVerbosity",
-        ), "TracingVerbosity must be NormalVerbosity or MaximalVerbosity"
+        if self.legacy_tracing:
+            assert self.node_config.get(
+                "TraceChainSyncClient", False
+            ), "TraceChainSyncClient not enabled"
+
+            assert self.node_config.get(
+                "TraceBlockFetchClient", False
+            ), "TraceBlockFetchClient not enabled"
+
+            # What are the other possible values? This should allow everything that is above Normal
+            assert self.node_config.get("TracingVerbosity", "") in (
+                "NormalVerbosity",
+                "MaximalVerbosity",
+            ), "TracingVerbosity must be NormalVerbosity or MaximalVerbosity"
+        else:
+            def checkCfg(target, level, inheritPath, cfg):
+                details = cfg.get("details", None)
+                maxFrequency = cfg.get("maxFrequency", None)
+                severity = cfg.get("severity", None)
+
+                error = False
+
+                if details not in ["DNormal", "DDetailed", "DMaximum"]:
+                    logger.error(f"For tracer '{target}' the {level} was found, but details of '{details}' is not DNormal, DDetailed, DMaximum")
+                    error = True
+
+                if maxFrequency not in [0, None]:
+                    logger.error(f"For tracer '{target}' the {level} was found, but maxFrequency of '{maxFrequency}' is not 0.0 or undeclared")
+                    error = True
+
+                if severity not in ["Info", "Debug"]:
+                    logger.error(f"For tracer '{target}' the {level} was found, but severity of '{severity}' is not Info or Debug")
+                    error = True
+
+                if level != "target":
+                    logger.warning(f"Ideal config for TraceOptions tracer '{target}' is to declare it rather than inherit from {level} of '{inheritPath}' with:")
+                    logger.warning(f'{{"{target}":{{"details":"DNormal","maxFrequency":0.0,"severity":"Info"}}')
+                elif error:
+                    logger.error(f"Ideal config for TraceOptions tracer '{target}' is to declare it explicitly with:")
+                    logger.error(f'{{"{target}":{{"details":"DNormal","maxFrequency":0.0,"severity":"Info"}}')
+
+                if error:
+                    sys.exit()
+
+            requiredTraceOptions = [
+                "BlockFetch.Client.CompletedBlockFetch",
+                "BlockFetch.Client.SendFetchRequest",
+                "ChainDB.AddBlockEvent.AddedToCurrentChain",
+                "ChainDB.AddBlockEvent.SwitchedToAFork",
+                "ChainSync.Client.DownloadedHeader"
+            ]
+
+            if self.node_config.get("UseTraceDispatcher", {}) != True:
+                logger.error('The legacy tracing system appears to be in use as "UseTraceDispatcher" is not declared true')
```

Comment:
`UseTraceDispatcher: true` is the default (see below), so it can be omitted. It might be more feasible to test for an explicit `UseTraceDispatcher: false` and conclude that legacy tracing is being used.

https://github.com/IntersectMBO/cardano-node/blob/76eeba1dfee4e5f90a11667b5fdcffe0dfb9d803/cardano-node/src/Cardano/Node/Configuration/POM.hs#L341

Reply:
Good point, thanks!  Updated in 6f213bf